packages/next/taskfile.js

```diff
@@ -2835,11 +2835,21 @@ export async function build(task, opts) {
 }
 
 export async function generate_types(task, opts) {
-  await execa(
+  const watchmode = opts.dev
+  const typesPromise = execa(
     'pnpm',
-    ['run', 'types', ...(opts.dev ? ['--watch', '--preserveWatchOutput'] : [])],
+    [
+      'run',
+      'types',
+      ...(watchmode ? ['--watch', '--preserveWatchOutput'] : []),
+    ],
     { stdio: 'inherit' }
   )
+  // In watch-mode the process never completes i.e. the Promise never resolve.
+  // But taskr needs to know that it can start watching the files for the task it has to manually restart.
+  if (!watchmode) {
+    await typesPromise
+  }
```

Comment:
Ideally we'd await that the process actually spawned in dev. Let's just hope `tsc` almost always starts even with tsconfig issues.